### PR TITLE
Analyzer: Add support for Poetry package manager

### LINKED ISSUE #1699: Add support for the Poetry (Python) package manager
Add support for the Poetry dependency manager to the analyzer module.

package manager website: https://github.com/sdispater/poetry

### PR DESCRIPTION
**Pull request by qequ**
Added a package manager class Poetry to handle Python's projects that use Poetry and functional tests 

Fixes #1699

### CODE REVIEW

File: analyzer/src/funTest/kotlin/managers/PoetryFunTest.kt
Comment (mnonnenmacher) [inline analyzer/src/funTest/kotlin/managers/PoetryFunTest.kt:34]: You need install Poetry in the Dockerfile, otherwise the test case cannot succeed on CI.

File: analyzer/src/funTest/assets/projects/synthetic/poetry/pyproject.toml
Comment (mnonnenmacher) [inline analyzer/src/funTest/assets/projects/synthetic/poetry/pyproject.toml:36]: It would be good to reduce the amount of dependencies in the test project. This would reduce the size of the expected result file and speed up the test.

File: analyzer/src/funTest/kotlin/managers/PoetryFunTest.kt
Comment (mnonnenmacher) [inline analyzer/src/funTest/kotlin/managers/PoetryFunTest.kt:2]: Wrong copyright statement.

File: analyzer/src/funTest/assets/projects/synthetic/poetry-expected-output.yml
Comment (mnonnenmacher) [inline analyzer/src/funTest/assets/projects/synthetic/poetry-expected-output.yml:686]: The expected output files should use the dependency tree representation instead of the dependency graph representation, because it is human readable. If you generate the file you can use the function `withResolvedScopes()` on the `AnalyzerResult` class to get the dependency tree representation before serialization. Compare for example with `pip-expected-output.yml`.

File: analyzer/src/funTest/assets/projects/synthetic/poetry-expected-output.yml
Comment (mnonnenmacher) [inline analyzer/src/funTest/assets/projects/synthetic/poetry-expected-output.yml:2]: The expected output file should be a serialized `ProjectAnalyzerResult`, not a whole `OrtResult` as this contains irrelevant information like the `environment` section below.

File: analyzer/src/funTest/assets/projects/synthetic/poetry-expected-output.yml
Comment (mnonnenmacher) [inline analyzer/src/funTest/assets/projects/synthetic/poetry-expected-output.yml:45]: `vcs_processed` should be:

```yaml
  vcs_processed:
    type: "Git"
    url: "<REPLACE_URL>"
    revision: "<REPLACE_REVISION>"
    path: "<REPLACE_PATH>"
```

**Review by mnonnenmacher — CHANGES_REQUESTED** [review]
@qequ Thank you for the contribution! Please see my inline comments and make sure to run `./gradlew detekt` to check for formatting issues.

File: analyzer/src/main/kotlin/managers/Poetry.kt
Comment (qequ) [reply analyzer/src/main/kotlin/managers/Poetry.kt:2]: @mnonnenmacher how should I treat the copyright? update the year? do you have any examples to look at?

Comment (mnonnenmacher) [reply analyzer/src/main/kotlin/managers/Poetry.kt:2]: The year must be 2022, and you are not making this contribution in the name of HERE Europe B.V., so if this is a private contribution the statement should be your name, otherwise the name of the company you are doing this contribution for.

**Comment by sschuberth** [comment]
Just a general remark @qequ: Please address review comments in the actual commit the review comment refers to by fixing up the commit and force-pushing a new version of your branch, instead of adding new commits that address review comments separately.

File: analyzer/src/main/kotlin/managers/Poetry.kt
Comment (mnonnenmacher) [inline analyzer/src/main/kotlin/managers/Poetry.kt:65]: This should be indented by four more space, I'm surprised that detekt does not complain, but maybe it's already fixed in a later commit?

**Review by mnonnenmacher — CHANGES_REQUESTED** [review]
@qequ Thanks for addressing the review comments. As @sschuberth already mentioned, please fix the original commits (by using Git's interactive rebase) instead of adding new commits. In this case you could also squash everything into a single commit, as adding the package manager and the tests for it don't have to be in separate commits.

**Comment by qequ** [comment]
@mnonnenmacher I will squash everything in a single commit

**Comment by mnonnenmacher** [comment]
> @mnonnenmacher I will squash everything in a single commit

Please fixup the commit message, it now has your sign-off statement several times and all the messages from the squashed commits which are now irrelevant after squashing. While at it, you could align the title with the title of this PR and make sure that all sentences end with a dot.

**Comment by qequ** [comment]
@mnonnenmacher fixed commit message and title

**Review by mnonnenmacher — APPROVED** [review]

**Comment by sschuberth** [comment]
> Fixes #1699.

One more nit: This should actually go into the commit message, not the PR discussion, to make the issue close automatically once the PR is merged.

**Comment by mnonnenmacher** [comment]
@sschuberth I saw your comment too late, but at least the issue was still closed automatically.

**Comment by sschuberth** [comment]
> but at least the issue was still closed automatically.

Yeah, good to know it also works if the reference is just in the PR discussion.

**Pull request merged**
